### PR TITLE
api list view for Enterprise Learner Engagements

### PR DESCRIPTION
**Pull request by muhammad-ammar**
**JIRA:** [ENT-2548](https://openedx.atlassian.net/browse/ENT-2548)

### CODE REVIEW

File: analytics_data_api/v0/serializers.py [hunk 1]
```diff
@@ -449,6 +449,12 @@ def get_videos_viewed(self, obj):
         return obj.get('videos_viewed', 0)
 
 
+class EnterpriseLearnerEngagementSerializer(serializers.ModelSerializer):
+    class Meta(object):
+        model = models.ModuleEngagement
+        fields = '__all__'
```
Comment (irfanuddinahmad) [inline analytics_data_api/v0/serializers.py]: Do we need all fields here? The ticket mentions the following:
course_id
username
date
event
count
entity_type

File: analytics_data_api/v0/views/learners.py [hunk 2]
```diff
@@ -352,6 +355,22 @@ def get_queryset(self):
...
+    serializer_class = EnterpriseLearnerEngagementSerializer
+    pagination_class = EdxPaginationSerializer
+
+    def get_queryset(self):
+        enterprise_users = EnterpriseUser.objects.filter(
+            enterprise_id=self.kwargs.get('enterprise_customer')
+        ).values_list(
+            'user_username', flat=True
+        )
+        return ModuleEngagement.objects.filter(username__in=enterprise_users)
```
Comment (ziafazal) [inline analytics_data_api/v0/views/learners.py]: is there any way we can avoid this `__in` operation? [might be costly](https://dba.stackexchange.com/questions/207255/performance-mysql-id-vs-id-in-multiple-values)

File: analytics_data_api/v0/urls/learners.py [hunk 3]
```diff
@@ -14,4 +16,6 @@
         views.EngagementTimelineView.as_view(), name='engagement_timelines'),
     url(r'^course_learner_metadata/{}/$'.format(COURSE_ID_PATTERN),
         views.CourseLearnerMetadata.as_view(), name='course_learner_metadata'),
+    url(r'^enterprise_learner_engagements/(?P<enterprise_customer>{})/$'.format(UUID_REGEX_PATTERN),
+        views.EnterpriseLearnerEngagementView.as_view(), name='enterprise_learner_engagements'),
```
[the remark this answers is not in the document]
Comment (muhammad-ammar) [reply analytics_data_api/v0/urls/learners.py]: agree that `engagements` is better but this endpoint will always return enterprise related data, that is why I named it like this.

[the remark this answers is not in the document]
Comment (muhammad-ammar) [reply analytics_data_api/v0/urls/learners.py]: updated

File: analytics_data_api/v0/serializers.py (hunk 1 above)
Comment (muhammad-ammar) [reply analytics_data_api/v0/serializers.py]: api consumer can pick the interested fields and ignore the rest.

**Comment by muhammad-ammar** [comment]
@irfanuddinahmad I have filtered the `discussion` types.

File: analytics_data_api/v0/urls/learners.py [hunk 4]
```diff
@@ -14,4 +16,6 @@
         views.EngagementTimelineView.as_view(), name='engagement_timelines'),
     url(r'^course_learner_metadata/{}/$'.format(COURSE_ID_PATTERN),
         views.CourseLearnerMetadata.as_view(), name='course_learner_metadata'),
+    url(r'^engagements/(?P<enterprise_customer>{})/$'.format(UUID_REGEX_PATTERN),
```
Comment (irfanuddinahmad) [inline analytics_data_api/v0/urls/learners.py]: Have a look at https://github.com/edx/edx-enterprise-data/pull/173/files#diff-986e81b28b1b96da14ef4a454b1ec752R122
The url should be /enterprise/{enterprise_uuid}/engagements/

Comment (muhammad-ammar) [reply analytics_data_api/v0/urls/learners.py]: I have made the URL format like existing urls in this repo.

**Pull request merged**